### PR TITLE
Rework platform

### PR DESCRIPTION
**Pull request by daniel-j**
Here is a preview:
screenshot
This rework of platform in runners makes it easier to filter games by platform and manufacturer. Multi-platform runners (such as libretro) convert the selected core to a platform valaue that appears in the sidebar and in the list view column. For example Linux games appear both in platform "PC" and "PC Linux".

### CODE REVIEW

File: lutris/runners/dosbox.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def makeconfig(path, drives, commands):
 class dosbox(Runner):
     human_name = "DOSBox"
     description = "MS-Dos emulator"
-    platform = "MS-DOS"
+    platforms = "MS-DOS"
```
Comment (RobLoach) [inline lutris/runners/dosbox.py]: Should MS-DOS appear as `('PC', 'DOS')`?

File: lutris/runners/scummvm.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 class scummvm(Runner):
     description = "Runs various 2D point-and-click adventure games."
     human_name = "ScummVM"
-    platform = "2D point-and-click games"
+    platforms = "2D point-and-click games"
```
Comment (RobLoach) [inline lutris/runners/scummvm.py]: ScummVM should likely be `PC`, the majority of the games were originally released on DOS.

File: lutris/runners/residualvm.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 class residualvm(Runner):
     human_name = "ResidualVM"
-    platform = "3D point-and-click games"
+    platforms = "3D point-and-click games"
```
Comment (RobLoach) [inline lutris/runners/residualvm.py]: Grim Fandango, Escape from Monkey Isaldn and Myst 3 were relased for `('PC', 'Windows')`.

File: lutris/runners/openmsx.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 class openmsx(Runner):
     human_name = "openMSX"
     description = "MSX computer emulator"
-    platform = "MSX, MSX2, MSX2+, MSX turboR"
+    platform = "MSX, MSX2, MSX2+, MSX turboR" # TODO
```
Comment (RobLoach) [inline lutris/runners/openmsx.py]: ?

File: lutris/runners/mupen64plus.py [hunk 5]
```diff
@@ -7,7 +7,7 @@
 class mupen64plus(Runner):
     human_name = "Mupen64Plus"
     description = "Nintendo 64 emulator"
-    platform = "Nintendo 64"
+    platforms = ('Nintendo', '64')
```
Comment (RobLoach) [inline lutris/runners/mupen64plus.py]: Should it be `('Nintendo', 'Nintendo 64')`? Or N64?

File: lutris/runners/mess.py [hunk 6]
```diff
@@ -6,7 +6,27 @@
...
     description = "Multi-system (consoles and computers) emulator"
-    platform = 'multi-platform'
+    # TODO: A lot of platforms/machines are missing
+    platforms = (
+        ('Amstrad', 'CPC 464'),
+        ('Amstrad', 'CPC 6128'),
+        ('Amstrad', 'GX4000'),
+        ('Apple', 'II'),
+        ('Apple', 'IIGS'),
+        ('Commodore 64',),
```
Comment (RobLoach) [inline lutris/runners/mess.py]: Should Commodore be `('Commodore', '64')`. There's a Commodore Amiga.

**Comment by RobLoach** [comment]
Looking great! Allows some nice filtering.

File: lutris/runners/openmsx.py (hunk 4 above)
Comment (daniel-j) [inline lutris/runners/openmsx.py]: openMSX is unused

File: lutris/runners/dosbox.py (hunk 1 above)
Comment (daniel-j) [inline lutris/runners/dosbox.py]: I don't want emulators to be in the PC filter, that's why I left it as MS-DOS.

**Pull request merged**